jsx/Panel.js

```diff
@@ -41,17 +41,55 @@ class Panel extends React.Component {
         "glyphicon pull-right glyphicon-chevron-up"
     );
 
+    let pencilGlyph;
+    if (this.props.edit) {
```

Comment:
if the prop passed here is a function, it would have to `return true` for the Glyph to be set. What happens when the function returns false by default?
I would suggest checking the function is set using `typeof` or something similar

Same comment for similar code below

Reply:
whatever it returns is irrelevant since this doesn't invoke the function, it just checks whether something was passed or not / isn't `null` or `undefined` etc.

Reply:
ah you're right, my mistake